src/proof/resolution_bitvector_proof.cpp

```diff
@@ -0,0 +1,426 @@
+/*********************                                                        */
+/*! \file resolution_bitvector_proof.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Liana Hadarean, Guy Katz, Paul Meng
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2018 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** [[ Add lengthier description here ]]
+
+ ** \todo document this file
+
+**/
+
+#include "proof/resolution_bitvector_proof.h"
+#include "proof/bitvector_proof.h"
+#include "options/bv_options.h"
+#include "options/proof_options.h"
+#include "proof/array_proof.h"
+#include "proof/clause_id.h"
+#include "proof/lfsc_proof_printer.h"
+#include "proof/proof_output_channel.h"
+#include "proof/proof_utils.h"
+#include "proof/sat_proof_implementation.h"
+#include "prop/bvminisat/bvminisat.h"
+#include "theory/bv/bitblast/bitblaster.h"
+#include "theory/bv/theory_bv.h"
+#include "theory/bv/theory_bv_rewrite_rules.h"
+
+using namespace CVC4::theory;
+using namespace CVC4::theory::bv;
+
+namespace CVC4 {
+
+ResolutionBitVectorProof::ResolutionBitVectorProof(theory::bv::TheoryBV* bv,
+                               TheoryProofEngine* proofEngine)
+    : BitVectorProof(bv, proofEngine),
+      d_resolutionProof(NULL),
+      d_cnfProof(NULL),
+      d_isAssumptionConflict(false)
+      {}
+
+void ResolutionBitVectorProof::initSatProof(CVC4::BVMinisat::Solver* solver) {
+  Assert (d_resolutionProof == NULL);
+  d_resolutionProof = new BVSatProof(solver, &d_fakeContext, "bb", true);
```

Comment:
It would be good to clarify the owners of `d_resolutionProof` and `d_cnfProof`. Right now, it looks like they are owned by `ResolutionBitVectorProof`. If they are, make them unique pointers, s.t. they are cleaned up when `ResolutionBitVectorProof` is destroyed (it looks like we are leaking them right now?).